state/forcedoperation.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import "time"
+
+// ForcedOperation that allows accumulation of operational errors and
+// can be forced.
+type ForcedOperation struct {
+	// Force controls whether or not the removal of a unit
+	// will be forced, i.e. ignore operational errors.
+	Force bool
+
+	// Errors contains errors encountered while applying this operation.
+	// Generally, these are non-fatal errors that have been encountered
+	// during, say, force. They may not have prevented the operation from being
+	// aborted but the user might still want to know about them.
+	Errors []error
+
+	// MaxWait specifies the amount of time that each step in relation destroy process
+	// will wait before forcing the next step to kick-off. This parameter
+	// only makes sense in combination with 'force' set to 'true'.
+	MaxWait time.Duration
+}
+
+// AddError adds an error to the collection of errors for this operation.
+func (op *ForcedOperation) AddError(one ...error) {
+	op.Errors = append(op.Errors, one...)
+}
+
+// FatalError returns true if the err is not nil and Force is false.
+// If the error is not nil, it's added to the slice of errors for the
+// operation.
+func (op *ForcedOperation) FatalError(err error) bool {
```

Comment:
Is there a better name?

Reply:
`CheckFatal` comes to mind, but I think this is OK.